ocaml/typing/typetexp.ml

```diff
@@ -426,6 +426,11 @@ let check_arg_type styp =
     | _ -> ()
   end
 
+let transl_label = function
+  | Parsetree.Labelled l -> Types.Labelled l
+  | Parsetree.Optional l -> Optional l
+  | Parsetree.Nolabel -> Nolabel
+
```

Comment:
I wasn't completely sure if this is the right place for this function.

Reply:
I like your choice of location, but I'm worried that most usages of this function will be wrong. Any place you're tempted to translate from an `Asttypes.arg_label` to a `Types.arg_label`, you probably need to check for `[%src_pos]` in some way.

I didn't worry about the use in ocamldoc, because that code is essentially deprecated. (It needs to compile, but it doesn't need to work perfectly.) But other spots are more worrisome. I'll comment more at usage sites.

Reply:
When adding Positions, would it make sense to make `transl_label` take in the argument type? I think it'd be a little nicer, but might result in checking for the `src_pos` extension redundantly

Reply:
Maybe? That will make the function hard to call for terms, where you'd have to extract the type from a pattern constraint. So maybe it becomes `transl_label : Parsetree.arg_label -> Parsetree.core_type option -> Types.arg_label`. That might work. And the `[%src_pos]` check should generally be very cheap, so I'm not worried about doing it redundantly.